Fix the tasksModule repository error

Context

The current implementation calls ```supabase.from("task")```, but ```"task"``` is not present in the generated Supabase TypeScript schema. Because of that:

TypeScript rejects ```from("task")```
query results are inferred as unrelated table unions
returned rows cannot be safely mapped into TaskRow
the repository fails compile-time checks across ```saveNew```, ```persistUpdates```, ```findById```, and ```findByUserId```

This likely means one of the following:

the task table does not exist in the actual database
the table exists but generated types are outdated
the repository is using the wrong table name

```
src/v1/modules/tasksModule/infrastructure/SupabaseTaskRepos
itory.ts:56:13 - error TS2769: No overload matches this cal
l.
    Argument of type '"task"' is not assignable to paramete
r of type 'never'.

56       .from(this.tableName)
               ~~~~~~~~~~~~~~

src/v1/modules/tasksModule/infrastructure/SupabaseTaskRepos
itory.ts:63:29

63     return this.mapToDomain(data);
                               ~~~~

src/v1/modules/tasksModule/infrastructure/SupabaseTaskRepos
itory.ts:70:13 - error TS2769: No overload matches this cal
l.

70       .from(this.tableName)
               ~~~~~~~~~~~~~~

src/v1/modules/tasksModule/infrastructure/SupabaseTaskRepos
itory.ts:78:29

78     return this.mapToDomain(data);
                               ~~~~
src/v1/modules/tasksModule/infrastructure/SupabaseTaskRepos
itory.ts:83:13 - error TS2769: No overload matches this cal
l.

83       .from(this.tableName)
               ~~~~~~~~~~~~~~

src/v1/modules/tasksModule/infrastructure/SupabaseTaskRepos
itory.ts:91:29 - error TS2345: Argument of type '{ created_
at:

91     return this.mapToDomain(data);
                               ~~~
src/v1/modules/tasksModule/infrastructure/SupabaseTaskRepos
itory.ts:103:13 - error TS2769: No overload matches this ca
ll.

103       .from(this.tableName)
                ~~~~~~~~~~~~~~

src/v1/modules/tasksModule/infrastructure/SupabaseTaskRepos
itory.ts:112:56 - error TS2345: Argument of type '{ created
_at:

112       list: (data || []).map((row) => this.mapToDomain(
row)),
                                                           
~~~

Found 8 errors in the same file, starting at: src/v1/module
s/tasksModule/infrastructure/SupabaseTaskRepository.ts:56
```

--- a/apps/identity-api/src/types/supabase.schema.ts
+++ b/apps/identity-api/src/types/supabase.schema.ts
@@ -597,6 +597,52 @@ export const publicStudyJamUpdateSchema = z.object({
   title: z.string().optional(),
 });
 
+export const publicTaskRowSchema = z.object({
+  completed_at: z.string().nullable(),
+  created_at: z.string().nullable(),
+  description: z.string().nullable(),
+  id: z.string(),
+  is_completed: z.boolean().nullable(),
+  name: z.string().nullable(),
+  points_on_completion: z.number().nullable(),
+  updated_at: z.string().nullable(),
+  user_id: z.string(),
+});
+
+export const publicTaskInsertSchema = z.object({
+  completed_at: z.string().optional().nullable(),
+  created_at: z.string().optional().nullable(),
+  description: z.string().optional().nullable(),
+  id: z.string().optional(),
+  is_completed: z.boolean().optional().nullable(),
+  name: z.string().optional().nullable(),
+  points_on_completion: z.number().optional().nullable(),
+  updated_at: z.string().optional().nullable(),
+  user_id: z.string(),
+});
+
+export const publicTaskUpdateSchema = z.object({
+  completed_at: z.string().optional().nullable(),
+  created_at: z.string().optional().nullable(),
+  description: z.string().optional().nullable(),
+  id: z.string().optional(),
+  is_completed: z.boolean().optional().nullable(),
+  name: z.string().optional().nullable(),
+  points_on_completion: z.number().optional().nullable(),
+  updated_at: z.string().optional().nullable(),
+  user_id: z.string().optional(),
+});
+
+export const publicTaskRelationshipsSchema = z.tuple([
+  z.object({
+    foreignKeyName: z.literal("task_user_id_fkey"),
+    columns: z.tuple([z.literal("user_id")]),
+    isOneToOne: z.literal(false),
+    referencedRelation: z.literal("user"),
+    referencedColumns: z.tuple([z.literal("id")]),
+  }),
+]);
+
 export const publicTeamRowSchema = z.object({
   description: z.string(),
   id: z.string(),

--- a/apps/identity-api/src/types/supabase.types.ts
+++ b/apps/identity-api/src/types/supabase.types.ts
@@ -574,6 +574,50 @@ export type Database = {
         }
         Relationships: []
       }
+      task: {
+        Row: {
+          completed_at: string | null
+          created_at: string | null
+          description: string | null
+          id: string
+          is_completed: boolean | null
+          name: string | null
+          points_on_completion: number | null
+          updated_at: string | null
+          user_id: string
+        }
+        Insert: {
+          completed_at?: string | null
+          created_at?: string | null
+          description?: string | null
+          id?: string
+          is_completed?: boolean | null
+          name?: string | null
+          points_on_completion?: number | null
+          updated_at?: string | null
+          user_id: string
+        }
+        Update: {
+          completed_at?: string | null
+          created_at?: string | null
+          description?: string | null
+          id?: string
+          is_completed?: boolean | null
+          name?: string | null
+          points_on_completion?: number | null
+          updated_at?: string | null
+          user_id?: string
+        }
+        Relationships: [
+          {
+            foreignKeyName: "task_user_id_fkey"
+            columns: ["user_id"]
+            isOneToOne: false
+            referencedRelation: "user"
+            referencedColumns: ["id"]
+          },
+        ]
+      }
       team: {
         Row: {
           description: string

--- a/apps/nexus-api/src/v1/loaders/loadRoutes.ts
+++ b/apps/nexus-api/src/v1/loaders/loadRoutes.ts
@@ -47,9 +47,7 @@ export const loadRoutes = (app: Express) => {
   const rolesRouter = new RolesRouter(rolesHttpController);
   const tasksHttpController = new TasksHttpController(taskModuleController);
   const tasksRouter = new TasksRouter(tasksHttpController);
-
-  const rolesHttpController=  new RolesHttpController(rbacController);
-  const rolesRouter=  new RolesRouter(rolesHttpController);
+ 
 
 
   const usersHttpController = new UsersHttpController(rbacController);

--- a/apps/nexus-api/src/v1/modules/tasksModule/infrastructure/SupabaseTaskRepository.ts
+++ b/apps/nexus-api/src/v1/modules/tasksModule/infrastructure/SupabaseTaskRepository.ts
@@ -1,19 +1,10 @@
+import { Tables } from "@/v1/types/supabase.types";
 import { ITaskRepository } from "../domain/ITaskRepository";
 import { Task, TaskProps } from "../domain/Task";
 import { supabase } from "@/v1/lib/supabase";
 import { handlePostgresError } from "@/v1/lib/supabase.utils";
 
-type TaskRow = {
-  id: string;
-  user_id: string;
-  name: string;
-  description: string;
-  points_on_completion: number;
-  is_completed: boolean;
-  completed_at: string | null;
-  created_at: string;
-  updated_at: string;
-};
+type TaskRow = Tables<"task">;
 
 type TaskInsertDTO = Omit<TaskRow, never>;
 
@@ -24,13 +15,13 @@ export class SupabaseTaskRepository implements ITaskRepository {
     return Task.hydrate({
       id: row.id,
       userId: row.user_id,
-      name: row.name,
-      description: row.description,
-      pointsOnCompletion: row.points_on_completion,
-      isCompleted: row.is_completed,
+      name: row.name || "",
+      description: row.description || "",
+      pointsOnCompletion: row.points_on_completion || 0,
+      isCompleted: row.is_completed || false,
       completedAt: row.completed_at ? new Date(row.completed_at) : null,
-      createdAt: new Date(row.created_at),
-      updatedAt: new Date(row.updated_at),
+      createdAt: new Date(row.created_at || Date.now()),
+      updatedAt: new Date(row.updated_at || Date.now()),
     });
   }
 

--- a/apps/nexus-api/src/v1/types/supabase.schema.ts
+++ b/apps/nexus-api/src/v1/types/supabase.schema.ts
@@ -597,6 +597,52 @@ export const publicStudyJamUpdateSchema = z.object({
   title: z.string().optional(),
 });
 
+export const publicTaskRowSchema = z.object({
+  completed_at: z.string().nullable(),
+  created_at: z.string().nullable(),
+  description: z.string().nullable(),
+  id: z.string(),
+  is_completed: z.boolean().nullable(),
+  name: z.string().nullable(),
+  points_on_completion: z.number().nullable(),
+  updated_at: z.string().nullable(),
+  user_id: z.string(),
+});
+
+export const publicTaskInsertSchema = z.object({
+  completed_at: z.string().optional().nullable(),
+  created_at: z.string().optional().nullable(),
+  description: z.string().optional().nullable(),
+  id: z.string().optional(),
+  is_completed: z.boolean().optional().nullable(),
+  name: z.string().optional().nullable(),
+  points_on_completion: z.number().optional().nullable(),
+  updated_at: z.string().optional().nullable(),
+  user_id: z.string(),
+});
+
+export const publicTaskUpdateSchema = z.object({
+  completed_at: z.string().optional().nullable(),
+  created_at: z.string().optional().nullable(),
+  description: z.string().optional().nullable(),
+  id: z.string().optional(),
+  is_completed: z.boolean().optional().nullable(),
+  name: z.string().optional().nullable(),
+  points_on_completion: z.number().optional().nullable(),
+  updated_at: z.string().optional().nullable(),
+  user_id: z.string().optional(),
+});
+
+export const publicTaskRelationshipsSchema = z.tuple([
+  z.object({
+    foreignKeyName: z.literal("task_user_id_fkey"),
+    columns: z.tuple([z.literal("user_id")]),
+    isOneToOne: z.literal(false),
+    referencedRelation: z.literal("user"),
+    referencedColumns: z.tuple([z.literal("id")]),
+  }),
+]);
+
 export const publicTeamRowSchema = z.object({
   description: z.string(),
   id: z.string(),

--- a/apps/nexus-api/src/v1/types/supabase.types.ts
+++ b/apps/nexus-api/src/v1/types/supabase.types.ts
@@ -574,6 +574,50 @@ export type Database = {
         }
         Relationships: []
       }
+      task: {
+        Row: {
+          completed_at: string | null
+          created_at: string | null
+          description: string | null
+          id: string
+          is_completed: boolean | null
+          name: string | null
+          points_on_completion: number | null
+          updated_at: string | null
+          user_id: string
+        }
+        Insert: {
+          completed_at?: string | null
+          created_at?: string | null
+          description?: string | null
+          id?: string
+          is_completed?: boolean | null
+          name?: string | null
+          points_on_completion?: number | null
+          updated_at?: string | null
+          user_id: string
+        }
+        Update: {
+          completed_at?: string | null
+          created_at?: string | null
+          description?: string | null
+          id?: string
+          is_completed?: boolean | null
+          name?: string | null
+          points_on_completion?: number | null
+          updated_at?: string | null
+          user_id?: string
+        }
+        Relationships: [
+          {
+            foreignKeyName: "task_user_id_fkey"
+            columns: ["user_id"]
+            isOneToOne: false
+            referencedRelation: "user"
+            referencedColumns: ["id"]
+          },
+        ]
+      }
       team: {
         Row: {
           description: string

